package.json

```diff
@@ -3,8 +3,7 @@
   "version": "1.1.0",
   "description": "Smart contracts for vanilladefi.com",
   "scripts": {
-    "preinstall": "npx only-allow pnpm",
-    "postinstall": "pnpm run generate:typechain",
+    "prepare": "pnpm run compile:sol && pnpm run generate:typechain",
```

Comment:
I'd like to keep the only-allow check, is it feasible to include in the prepare-phase?

Reply:
Actually it forces the package using contracts to use pnpm. Other scripts already explicitly use pnpm, so shouldn't be a problem

Reply:
For example it would force sdk to use pnpm, since that uses contracts (in my pr)

Reply:
But yeah, maybe we can use it in prepare, I overlooked that part 😅.

Reply:
Ok got it, so let's leave it out then :+1: 

And it's not a big problem, we pretty clearly document our pnpm usage anyway in README.md - I'm just not 100% sure what happens to `node_modules` and `package-lock.json` if you install first with pnpm and then npm :thinking:

Reply:
Oh,  if we want to force pnpm for installing deps, then even `prepare` won't help. Unlike mixing yarn/npm where they show warnings, there is no such warning in case of pnpm. Problem with mixing different package managers is missing dependencies, so those errors will show in build step.